app/views/layouts/_navbar.html.erb

```diff
@@ -1,9 +1,16 @@
 <%# https://getbootstrap.com/docs/4.5/components/navbar/ %>
 <nav class="navbar navbar-expand-md navbar-dark bg-dark">
   <div class="navbar-brand"><%= link_to I18n.t('index.portal_name'), root_path, class: 'portal-link' %></div>
-  <form class="form-inline">
-    <i class="fa fa-bell" style="color: white; padding: 10px"></i>
-    <a href="<%= search_users_path %>"><i class="fa fa-search" style="color: white; padding: 10px"></i> </a>
-  </form>
+  <% if user_signed_in? %>
+    <form class="form-inline">
+      <% if current_user.contact_requests.any? %>
+        <a href="<%= user_contact_requests_path(current_user) %>"> <i class="fa fa-bell" style="color: red; padding: 3px"></i> </a>
+        <span class="badge badge-pill badge-danger"> <%=  current_user.contact_requests.length %> </span>
```

Comment:
```suggestion
        <span class="badge badge-pill badge-danger"><%= current_user.contact_requests.length %></span>
```
This code is really hard to understand because the relation between notifications and contact requests is not really clear. I would prefer to have the logic in the model rather than in the controller so that I can just call `current_user.notification_count` in the view. The `if` statement has the same problem, maybe `current_user.notifications?` would be nice there (if we really want to hide the bell). This logic would then also need model specs.

Reply:
Good point!